spi_nand_flash/test_app/main/test_spi_nand_flash.c

```diff
@@ -153,14 +155,15 @@ static void do_single_write_test(spi_nand_flash_device_t *flash, uint32_t start_
 
 TEST_CASE("write nand flash sectors", "[spi_nand_flash]")
 {
-    uint16_t sector_num, sector_size;
+    uint32_t sector_num;
+    uint16_t sector_size;
     spi_nand_flash_device_t *nand_flash_device_handle;
     spi_device_handle_t spi;
     setup_nand_flash(&nand_flash_device_handle, &spi);
 
     TEST_ESP_OK(spi_nand_flash_get_capacity(nand_flash_device_handle, &sector_num));
     TEST_ESP_OK(spi_nand_flash_get_sector_size(nand_flash_device_handle, &sector_size));
```

Comment:
@UnTraDe Would it make sense to change the type also for `spi_nand_flash_get_sector_size`? I don't know if we will run into a part which has a 64kB sector size in the future. And it seems that there is no extra cost to using a 32-bit variable, since an arguments occupies the whole 32-bit register anyway.

Reply:
@igrr Yeah I also thought so at first, but then I saw this part: 
https://github.com/espressif/idf-extra-components/blob/2da4e71f7c5881df286a717e7a4c04fa25f08357/spi_nand_flash/diskio/diskio_nand.c#L95C1-L102C1

I tracked down `WORD` to be defined as 16bit. So it seems to be limited there, and allowing for `uint32_t` in a higher level API might introduce new bugs "silently" by having it truncated later. I also though about going the other way around and change the `page_size`in `spi_nand_flash_device_t` to be `uint16_t` but I haven't fully traced all the places that rely on that.

Reply:
Good point! What about expanding the sector size to a 32-bit variable and modifying the code in https://github.com/espressif/idf-extra-components/blob/2da4e71f7c5881df286a717e7a4c04fa25f08357/spi_nand_flash/diskio/diskio_nand.c#L99 to check for overflow before assigning? (And returning an error on overflow.) At least the underlying spi_nand_flash driver will be ready to support larger sector sizes, even if the upper layer FAT integration won't work.

Reply:
Yeah sounds good!